lib/e2e/src/deploy.rs

```diff
@@ -18,6 +18,7 @@ impl Deployer {
     }
 
     /// Add solidity constructor to the deployer.
+    #[allow(clippy::needless_pass_by_value)]
     pub fn with_constructor<C: SolConstructor + Send>(
```

Comment:
Decided to just ignore this warning, as addressing it would require updating all tests that use this function.
It's not that this would be hard, it's just that it would clutter the PR without adding too much benefit.

If you think it's worth it though, I'll remove this annotation and update the tests :+1:

Reply:
No need as it is only `e2e` lib :)